File: essentiel-app/src/components/WorkoutBarChart.js
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { Card } from 'react-native-paper';

const WorkoutBarChart = ({ data }) => {
    const maxValue = Math.max(...data); // Find the maximum value in the data array
    const range = Math.ceil(maxValue / 3); // Determine the range of y-axis values
    const yAxisValues = Array.from({ length: 4 }, (_, i) => i * range).reverse(); // Generate y-axis values

    // Check if all data values are zero
    const isNoData = data.every((value) => value === 0);

    return (
        <Card style={styles.card}>
            {isNoData ? (
                <Text style={styles.noDataText}>Nothing to see here</Text>
            ) : (
                <View style={styles.container}>
                    <View style={styles.yAxisContainer}>
                        {yAxisValues.map((value, index) => (
                            <Text key={index} style={styles.yAxisLabel}>
                                {value}
                            </Text>
                        ))}
                    </View>
                    {data.map((value, index) => (
                        <View
                            key={index}
                            style={styles.barContainer}
                        >
                            <View
                                style={[
                                    styles.bar,
                                    { height: `${(value / maxValue) * 76}%` },
                                ]}
                            />
                            <Text style={styles.label}>
                                {['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'][index]}
                            </Text>
                        </View>
                    ))}
                </View>
            )}
        </Card>
    );
};

const styles = StyleSheet.create({
    card: {
        padding: 10,
        backgroundColor: '#161616',
        borderColor: '#282828',
        borderWidth: 1,
        borderRadius: 24,
    },
    container: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        alignItems: 'flex-end',
        height: 150,
    },
    barContainer: {
        alignItems: 'center',
        position: 'relative',
    },
    bar: {
        width: 10,
        borderRadius: 5,
        backgroundColor: 'white',
        marginBottom: 5,
    },
    label: {
        color: 'white',
    },
    noDataText: {
        fontSize: 16,
        textAlign: 'center',
        padding: 10,
        color: 'lightgrey',
    },
    yAxisContainer: {
        justifyContent: 'space-between',
        height: 150,
        paddingBottom: 20,
        paddingTop: 8,
    },
    yAxisLabel: {
        color: 'white',
    },
});

export default WorkoutBarChart;

File: lib/components/WorkoutBarChart.dart
import 'package:flutter/material.dart';

class WorkoutBarChart extends StatelessWidget {
  final List<int> data;

  const WorkoutBarChart({Key? key, required this.data}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    int maxValue = data.isNotEmpty ? data.reduce((a, b) => a > b ? a : b) : 0;
    int range = maxValue > 0 ? (maxValue / 3).ceil() : 1;
    List<int> yAxisValues = List.generate(4, (i) => i * range).reversed.toList();
    bool isNoData = data.every((value) => value == 0);

    return Card(
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(24),
        side: const BorderSide(color: Color(0xFF282828), width: 1),
      ),
      color: const Color(0xFF161616),
      child: Padding(
        padding: const EdgeInsets.all(10),
        child: isNoData
            ? const Center(
                child: Text(
                  "Nothing to see here",
                  style: TextStyle(fontSize: 16, color: Colors.grey),
                ),
              )
            : Row(
                crossAxisAlignment: CrossAxisAlignment.end,
                children: [
                  Column(
                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                    children: yAxisValues
                        .map((value) => Text(
                              value.toString(),
                              style: const TextStyle(color: Colors.white),
                            ))
                        .toList(),
                  ),
                  const SizedBox(width: 10),
                  Expanded(
                    child: Row(
                      mainAxisAlignment: MainAxisAlignment.spaceAround,
                      children: List.generate(data.length, (index) {
                        double heightPercentage =
                            maxValue > 0 ? (data[index] / maxValue) * 76 : 0;
                        return Column(
                          children: [
                            Container(
                              width: 10,
                              height: heightPercentage,
                              decoration: BoxDecoration(
                                color: Colors.white,
                                borderRadius: BorderRadius.circular(5),
                              ),
                            ),
                            const SizedBox(height: 5),
                            Text(
                              ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]
                                  [index],
                              style: const TextStyle(color: Colors.white),
                            ),
                          ],
                        );
                      }),
                    ),
                  ),
                ],
              ),
      ),
    );
  }
}
